src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/requirethis/InputRequireThisRecordsWithCheckFields.java

```diff
@@ -0,0 +1,18 @@
+/*
+RequireThis
+checkFields = true
+checkMethods = (default)true
+validateOnlyOverlapping = (default)true
+
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.requirethis;
+
+public record InputRequireThisRecordsWithCheckFields(int a) {
+    void method() {
+        String a = "BUG";
+        a = a.substring(0, 1); // violation 'Reference to instance variable 'a' needs "this."'
```

Comment:
This should not be a violation, usage of `this` here would not compile due to different types (`int` vs `String`).

Reply:
I suggest to make any test input for records have the equivalent class also in the input, to make it easier to compare behavior between the two. Also, we should have the `// ok` equivalent as well.

Reply:
+1